lib/image_optim/cmd.rb

```diff
@@ -44,6 +80,42 @@ def check_status!
 
         fail SignalException, status.termsig
       end
+
+      def init_options!(args)
+        pgroup_opt = Gem.win_platform? ? :new_pgroup : :pgroup
+
+        if args.last.is_a?(Hash)
+          args.last[pgroup_opt] = true
+        else
+          args.push(pgroup_opt => true)
+        end
+      end
+
+      def cleanup_process(pid)
+        Thread.new do
```

Comment:
For every run of worker there will be maximum two tempfiles used (check `Handler`), so if you don't wait for cleanup, the process can be still writing to the tempfile while next worker will be using it.